### ISSUE TITLE
add `prefer-brand-last` rule

### ISSUE THREAD

**Issue by marcalexiei** [marcalexiei/eslint-plugin-zod-x#114]
### 🧩 Description

Similar to prefer-meta-last

### ⚙️ Options

N/A

### ✅ Valid Code Examples

```typescript
z.string().min(1).brand<'A'>()
```

### ❌ Invalid Code Examples

```typescript
z.string().brand<'A'>().min(1)
```
